### PR TITLE
Improving image syncing with salesforce

### PR DESCRIPTION
**Pull request by balazssimongen**
Linked, external images and relative images did not sync with Salesforce source. Now both can be used for syncing

### CODE REVIEW

File: src/salesforce/salesforce-adapter.ts [hunk 1]
```diff
@@ -44,6 +47,15 @@ export class SalesforceAdapter
     articleId: string | null,
     url: string,
   ): Promise<Image | null> {
+    if (!SalesforceAdapter.ATTACHMENT_RELATIVE_PATH_REGEX.test(url)) {
+      if (url.startsWith('/')) {
+        return Promise.resolve(null);
```
Comment (petergaras) [inline src/salesforce/salesforce-adapter.ts:52]: Personally I prefer returning null and making the method async which will wrap it into a Promise.
This Promise.resolve call is just noise.

File: src/salesforce/salesforce-loader.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ import { getLogger } from '../utils/logger.js';
 import { SalesforceArticleDetails } from './model/salesforce-article-details.js';
 import { AbstractLoader } from '../pipe/abstract-loader.js';
 import { SalesforceCategoryGroup } from './model/salesforce-category-group.js';
+import { isRelativeUrl } from '../utils/links';
```
Comment (petergaras) [inline src/salesforce/salesforce-loader.ts:12]: .js missing

File: src/salesforce/salesforce-loader.ts [hunk 3]
```diff
@@ -74,10 +75,23 @@ export class SalesforceLoader extends AbstractLoader {
...
-    const parsedUrl = new URL(url.replace(/&amp;/g, '&'));
+    let parsedUrl: URL;
+    try {
+      parsedUrl = new URL(url.replace(/&amp;/g, '&'));
+    } catch (error) {
+      // Invalid URL, treat it as relative
+      return url;
+    }
+
+    // const parsedUrl = new URL(url.replace(/&amp;/g, '&'));
```
Comment (petergaras) [inline src/salesforce/salesforce-loader.ts:86]: can be removed

File: src/salesforce/salesforce-loader.ts (hunk 2 above)
Comment (tkurucz-genesys) [inline src/salesforce/salesforce-loader.ts:12]: Missing file extension

File: src/salesforce/salesforce-loader.ts (hunk 3 above)
Comment (tkurucz-genesys) [inline src/salesforce/salesforce-loader.ts:86]: Do you still need this line?

File: src/salesforce/salesforce-loader.ts (hunk 2 above)
Comment (balazssimongen) [reply src/salesforce/salesforce-loader.ts]: Removed unused import

**Review by petergaras — APPROVED** [review]

File: src/salesforce/salesforce-adapter.ts (hunk 1 above)
Comment (balazssimongen) [reply src/salesforce/salesforce-adapter.ts]: Reworked and not matching relative urls and absolute urls are handled the same way now

**Review by tkurucz-genesys — APPROVED** [review]

**Pull request merged**